pic18f26k83/i2c.c

```diff
@@ -115,7 +115,18 @@ bool i2c_write_reg16(uint8_t address, uint8_t reg, uint16_t val) {
 }
 
 bool i2c_read_reg8(uint8_t address, uint8_t reg, uint8_t *value) {
-    return i2c_read_data(address, reg, value, 1);
+    bool success = i2c_write(address, &reg, 1);
```

Comment:
Add some comments with a link to the data sheet of a relevent IC explaining what this protocol is and why its needed